log/mock.go

```diff
@@ -18,6 +19,13 @@ func NewMock() *Mock {
 	return &Mock{impl{newMockWriter()}}
 }
 
+// NewWaitingMock creates a mock logger implementing the writer interface.
+// It stores all logged messages in a buffer for inspection by test
+// functions.
```

Comment:
This doccomment is not quite right since NewWaitingMock doesn't store messages in a buffer - it just writes them to an unbuffered channel.

Also, one thing that occurred to me: This "waiting mock" will only work when all the logging is happening in a different goroutine from the test. Can you document that here? There are some ways to potentially fix it but I don't think we need to handle them in this PR. We can make improvements later if we need to use this mock in a context where some log lines are written from the test thread and some are written from a different thread.

Reply:
The waitingMockWriter now is buffered, but I have shortened the comment anyway. With the buffer it has been tested to work when logging from the test goroutine.